src/utils/securityUtils.ts

```
@@ -0,0 +1,325 @@
+ * @example
+ * ```typescript
+ * const safe = sanitizeFilename('my<file>name.txt'); // Returns: 'my_file_name.txt'
+ * ```
+ */
+export function sanitizeFilename(filename: string): string {
+  // Remove null bytes
+  let sanitized = filename.replace(/\0/g, '');
+
+  // Replace dangerous characters with underscores
+  // Dangerous: < > : " / \ | ? * and control characters
+  sanitized = sanitized.replace(/[<>:"/\\|?*\x00-\x1F]/g, '_');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)